### DISCUSSION TITLE
Errors in route:list command when run from Artisan Facade

### DISCUSSION THREAD

**Discussion by zavre-c** [laravel/framework#36773] [Q&A]
### Errors while executing route:list command when run from Artisan Facade

Laravel version: 8.28.1
Laravel config cached: true
PHP version: 7.4.15

Steps to reproduce 
1. Through Web Request 
```php
    Route::get('/',function() {
        $exit_code = Artisan::call('route:list -c --json');
        $output = Artisan::output();
        return response($output,200,['Content-type':'application/json']);
    );

```

Error : Target class [web] does not exist.

While terminating
`$kernel->terminate($request,$response)`

while terminating middleware array of middlewares consist 'web' , otherwise it isn't
`$middlewares`

2. Through artisan tinker
`php artisan tinker`

```php
Artisan::call('route:list -c --json');
```
Error: PHP Warning:  class_implements(): Class web does not exist and could not be loaded

**Note:** In both above cases ```php shell_exec("php artisan route:list -c --json")```  and other artisan command `Artisan::call('inspire')` works 

_Ref:_ https://laracasts.com/discuss/channels/laravel/get-output-of-fired-artisan-command-from-route

**Comment by 7dir** [comment]
+1
